pkg/resource/managed_reconciler.go

```diff
@@ -432,6 +432,10 @@ func (r *ManagedReconciler) Reconcile(req reconcile.Request) (reconcile.Result,
 	ctx, cancel := context.WithTimeout(context.Background(), managedReconcileTimeout)
 	defer cancel()
 
+	// The status update call should have its own context in order to be able to
+	// report error in the case where the main context's deadline is exceeded.
+	updateCtx := context.Background()
```

Comment:
I am not sure that we want to use a context that will never timeout. In my mind it seems like it would make sense to say if our managed context timed out that we should use a new context with a relatively short timeout to try and update that status to reflect that, but it feels like an anti-pattern to use a context with no timeout in response to a previous context timing out.